### PR TITLE
[stable/datadog] fix daemonset templates for go 1.14

### PR DESCRIPTION
**Pull request by shunyy**
Signed-off-by: Shun Yamamoto <shun-yamamoto@m3.com>

#### Is this a new chart
No

#### What this PR does / why we need it:
- Current daemonset templates is not compatible with go 1.14, this PR fix that
  - related: https://github.com/helm/charts/pull/21888

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fieldas.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/datadog/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Datadog changelog
 
+## 2.3.30
+* Fixed daemoset template for go 1.14
```
Comment (nesl247) [inline stable/datadog/CHANGELOG.md]:
```suggestion
* Fixed daemonset template for go 1.14
```

**Review by nesl247 — CHANGES_REQUESTED** [review]

File: stable/datadog/CHANGELOG.md (hunk 1 above)
Comment (shunyy) [reply stable/datadog/CHANGELOG.md]: fixed!

**Comment by vboulineau** [comment]
Sorry, @shunyy You need to bump again to `.31`, got a PR merged in between

**Comment by shunyy** [comment]
@vboulineau 
I bumped version to 2.3.31！

**Pull request merged**